lms/data_tasks/report/create_from_scratch/02_entities/04_assignments/01_create_view.sql

```diff
@@ -0,0 +1,42 @@
+DROP MATERIALIZED VIEW IF EXISTS report.assignments CASCADE;
+
+-- A slightly sanitized copy of our assignments
+
+CREATE MATERIALIZED VIEW report.assignments AS (
+    WITH
+        assignments AS (
+            SELECT
+                id,
+                created,
+                updated,
+                -- Do some basic normalization
+                TRIM(REPLACE(LOWER(document_url), '%3a%2f%2f', '://')) AS url
+            FROM assignment
+        )
+
+    SELECT
+        *,
+        CASE
+            -- This happens some times for reasons unknown
+            WHEN url IS NULL OR url = '' THEN 'No Document'
+
+            -- Content integrations
+            WHEN STARTS_WITH(url, 'blackboard://') THEN 'Blackboard Files'
+            WHEN STARTS_WITH(url, 'd2l://') THEN 'D2L Files'
+            WHEN STARTS_WITH(url, 'canvas://') THEN 'Canvas Files'
+            WHEN STARTS_WITH(url, 'vitalsource://') THEN 'VitalSource'
+            WHEN STARTS_WITH(url, 'jstor://') THEN 'JSTOR'
+            WHEN STARTS_WITH(url, 'https://drive.google.com') THEN 'Google Drive'
+            WHEN STARTS_WITH(url, 'https://api.onedrive.com') THEN 'Microsoft OneDrive'
+            WHEN STARTS_WITH(url, 'https://www.youtube.com/watch') THEN 'YouTube'
```

Comment:
I'd exclude this for now, youtube can't be proxied by via right now and the new youtube functionally is unlikely to use these URLs directly.